app/controllers/discourse_sift/admin_mod_queue_controller.rb

```diff
@@ -14,68 +15,82 @@ def index
-
-        #Notify User?
-        if SiteSetting.sift_notify_user
-          SystemMessage.new(post.user).create(
-            'sift_has_moderated',
-            topic_title: post.topic.title
-          )
+      if should_use_reviewable_api?
+        reviewable.perform(current_user, :confirm_failed)
+      else
+        # If post has not been deleted (i.e. if setting is on)
+        # Then delete it now
+        if !post.deleted_at
+
+          #Rails.logger.error("sift_debug: Post not deleted.  Deleting now")
+
+          PostDestroyer.new(current_user, post).destroy
+
+          #Notify User?
+          if SiteSetting.sift_notify_user
+            SystemMessage.new(post.user).create(
+              'sift_has_moderated',
+              topic_title: post.topic.title
+            )
+          end
         end
+
+        DiscourseSift.move_to_state(post, 'confirmed_failed')
+        log_confirmation(post, 'confirmed_failed')
       end
 
-      
-      DiscourseSift.move_to_state(post, 'confirmed_failed')
-      log_confirmation(post, 'confirmed_failed')
       render body: nil
     end
 
     def allow
-      post = Post.with_deleted.find(params[:post_id])
+      if should_use_reviewable_api?
+        reviewable.perform(current_user, :allow)
+      else
+        # It's possible the post was recovered already
+        if post.deleted_at
+          PostDestroyer.new(current_user, post).recover
+        end
 
-      # It's possible the post was recovered already
-      if post.deleted_at
-        PostDestroyer.new(current_user, post).recover
+        DiscourseSift.move_to_state(post, 'confirmed_passed')
+        log_confirmation(post, 'confirmed_passed')
       end
 
-      DiscourseSift.move_to_state(post, 'confirmed_passed')
-      log_confirmation(post, 'confirmed_passed')
-
       render body: nil
     end
 
     def dismiss
-      post = Post.with_deleted.find(params[:post_id])
-
-      DiscourseSift.move_to_state(post, 'dismissed')
-      log_confirmation(post, 'dismissed')
+      if should_use_reviewable_api?
+        reviewable.perform(current_user, :ignore)
+      else
+        DiscourseSift.move_to_state(post, 'dismissed')
+        log_confirmation(post, 'dismissed')
+      end
 
       render body: nil
     end
 
     private
 
     def log_confirmation(post, custom_type)
-      topic = post.topic || Topic.with_deleted.find(post.topic_id)
-
-      StaffActionLogger.new(current_user).log_custom(custom_type,
-                                                     post_id: post.id,
-                                                     topic_id: topic.id,
-                                                     created_at: post.created_at,
-                                                     topic: topic.title,
-                                                     post_number: post.post_number,
-                                                     raw: post.raw
+      StaffActionLogger.new(current_user).log_custom(
```

Comment:
This seems to be logging a lot less. What's the reasoning?

Reply:
Those attributes are being filtered by the `StaffActionLogger`

Reply:
Thanks for the clarification.